### PR TITLE
Use chdir before executing svn command

### PR DESCRIPTION
**Pull request by GuillaumeSmaha**
Permit to execute request like :

    repo.list(False, '^/tags')

### CODE REVIEW

**Comment by GuillaumeSmaha** [comment]
I am fixing some issues. Wait before review

**Comment by GuillaumeSmaha** [comment]
Ok, I edit the test to add the main directories (trunk, tags and branches) in local repository and I add a test to list the tags directory with `repo.list(False, '^/tags')` from a local checkout on the `trunk` directory

**Comment by GuillaumeSmaha** [comment]
I forgot that svn 1.6 have a `.svn` directory in each directory.

**Comment by GuillaumeSmaha** [comment]
I squash the commits

**Comment by GuillaumeSmaha** [comment]
@tusharmakkar08 Any news to merge this PR ?

File: svn/common.py [hunk 1]
```diff
@@ -51,7 +54,17 @@ def run_command(self, subcommand, args, **kwargs):
...
         cmd += [subcommand] + args
-        return self.external_command(cmd, environment=self.__env, **kwargs)
+
+        if self.__type == svn.constants.LT_PATH:
+            cwd = os.getcwd()
+            os.chdir(self.__url_or_path)
+
+        result = self.external_command(cmd, environment=self.__env, **kwargs)
+
+        if self.__type == svn.constants.LT_PATH:
```
Comment (tusharmakkar08) [inline svn/common.py]: Can we have this in `try..finally` block ?

Comment (GuillaumeSmaha) [inline svn/common.py]: Thank's for the review. Do you need a squash of the commits ?

**Comment by tusharmakkar08** [comment]
Yes, preferably. 

@dsoprea : Please look into this PR as well. 

Thanks.

**Comment by GuillaumeSmaha** [comment]
@tusharmakkar08 I replace ` finally` by `except` and use `raise` to re-throw the exception.
https://github.com/dsoprea/PySvn/pull/85/commits/90a261f6970cc9f327bd6ee3e3876a2d6c17347b#diff-557e034a141260ba6a3cd24b65e0f8cdR67

And I add a test to test this exception : 
https://github.com/dsoprea/PySvn/pull/85/commits/90a261f6970cc9f327bd6ee3e3876a2d6c17347b#diff-1d23cefe300b7c6f8845d87f328366adR185

**Comment by GuillaumeSmaha** [comment]
@tusharmakkar08 Ok I didn't know that `finally` also calls `raise`. I update to keep ` finally`

**Comment by tusharmakkar08** [comment]
@GuillaumeSmaha : Please fix the failing test cases.

**Comment by GuillaumeSmaha** [comment]
@tusharmakkar08 I don't know why, but sometimes the checkout doesn't work. I had the same issue on my computer before push it.

**Comment by GuillaumeSmaha** [comment]
@tusharmakkar08 I fixed the issue.

**Comment by GuillaumeSmaha** [comment]
@tusharmakkar08 Any news to merge it ?

**Comment by tusharmakkar08** [comment]
@dsoprea : Please look into this.

**Comment by GuillaumeSmaha** [comment]
@dsoprea Hi ! Could you review this PR ?

**Comment by GuillaumeSmaha** [comment]
@dsoprea Could you review this PR ?

**Comment by dsoprea** [comment]
Please fix the conflicts.

**Comment by dsoprea** [comment]
Closed due to stale conflicts.

**Pull request closed without merging**